### DISCUSSION TITLE
How to use ghidra for remote debugging?

### DISCUSSION THREAD

**Discussion by SirEnder125** [NationalSecurityAgency/ghidra#2652] [Debugger]
Hello. I recently downloaded ghidra to try to debug 3ds games and make cheats.
Can anyone tell me how to use remote debugging with ghidra?

**Comment by d-millar** [comment]
If 3ds includes a gdb stub compatible with 7.0 or greater, you can use the "GNU gdb over GADP" interface.

**Comment by SirEnder125** [comment]
Uhhh I searched and I cannot find that.
Is it an extension for ghidra?
Can I have a link?

**Comment by d-millar** [comment]
Hmmm, well, from a Ghidra perspective, you're already there.  To be clearer, you'll either need a program on the 3ds to act as a gdbserver (a gdbstub), something like Luma3ds, or a 3ds emulator w/ the equivalent, something like Citra.  Once you have that, I would verify that you can connect via a current version of gdb.  If you can, you have some shot of connecting with Ghidra.  After that, I guess I would make sure you can build Ghidra by following the DevGuide.md provided.  If everything is still good, you can switch to the debugger branch, build using the DebuggerDevGuide.md, and connect using one of the gdb connectors.

**Comment by SirEnder125** [comment]
Okay, so I tried downloading gdb latest version and I can't build it. I can't figure out how.

**Comment by d-millar** [comment]
You can download the most recent official release of GDB from either Project GNU's FTP server, or Red Hat's sources site:

http://ftp.gnu.org/gnu/gdb (mirrors)
ftp://sourceware.org/pub/gdb/releases/ (mirrors).

**Comment by SirEnder125** [comment]
I'm sorry, I was not clear. I meant I downloaded the source and could not successfully build it.
I don't know why it gives the source, I want the .exe!

**Comment by d-millar** [comment]
https://sourceforge.net/projects/mingw/files/

**Comment by SirEnder125** [comment]
I could only find old version at that link. I have v8.3.1 currently.

**Comment by SirEnder125** [comment]
Okay, I downloaded the latest version from http://ftp.gnu.org/gnu/gdb and it ALMOST built.
I got the following error:

```
*** BFD does not support target x86_64-pc-msys.
*** Look in bfd/config.bfd for supported targets.
make[1]: *** [Makefile:2715: configure-bfd] Error 1
```

What is causing this?

**Comment by d-millar** [comment]
I'm confused - what architecture are you building on/for?

**Comment by SirEnder125** [comment]
I'm on Windows 10? x84???
I'm not sure what architecture is in computer terms.

**Comment by d-millar** [comment]
yeah, that - just making sure.  are you compiling from the mingw64 shell?  (I do think you could save yourself a LOT of work by downloading a reasonably current version of MingW64 with the executable pre-compiled.  Compiling gdb on Windows is really not for the faint of heart.)

**Comment by pplealf** [comment]
My laptop with Parrot and Kali, has been hacked. Help.

**Reply by SirEnder125** [reply]
Huh?!

**Comment by ndunks** [comment]
Are you successfully creating cheat ? :smile:

**Reply by SirEnder125** [reply]
Yep! A few, that is. I had great help from various sources, including this conversation, and I made a utility for no bead-head in game, for players who haven't played in a while. Although the game is pretty much dead at this point, I only know of one other developer at this point, so I haven't been working of this much lately.
